routers/web/admin/pprof.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package admin
+
+import (
+	"fmt"
+	"net/http"
+	"runtime/pprof"
+	"runtime/trace"
+	"strconv"
+	"time"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/process"
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/felixge/fgprof"
+)
+
+// PProfProcessStacktrace returns the stacktrace similar to GoroutineStacktrace but without rendering it
+func PProfProcessStacktrace(ctx *context.Context) {
+	flat := ctx.FormBool("flat")
+	noSystem := ctx.FormBool("no-system")
+
+	format := ctx.FormString("format")
+	jsonFormat := format == "json"
```

Comment:
should we perhaps compare with `strings.ToLower(format)`?

Reply:
These are technical endpoints that are internal only. They're not API.